pennylane_lightning/src/bindings/Bindings.cpp

```diff
@@ -324,6 +324,28 @@ void lightning_class_bindings(py::module &m) {
                 const std::vector<size_t> &wires) {
                  return M.expval(operation, wires);
              })
+        .def("generate_samples",
+             [](Measures<PrecisionT> &M, size_t num_shots) {
+                 auto &&result = M.generate_samples(num_shots);
+                 // int num_samples = num_shots;
+                 // std::vector<int> result(num_samples*wires.size(),0);
+
+                 const size_t num_wires = result.size() / num_shots;
+                 const size_t ndim = 2;
+                 const std::vector<size_t> shape{num_shots, num_wires};
+                 const std::vector<size_t> strides{sizeof(size_t) * num_wires,
+                                                   sizeof(size_t)};
+
+                 // return 2-D NumPy array
+                 return py::array(py::buffer_info(
+                     result.data(),  /* data as contiguous array  */
+                     sizeof(size_t), /* size of one scalar        */
```

Comment:
I would suggest something like:
```suggestion
                 constexpr auto sz = sizeof(size_t); 
                 const std::vector<size_t> strides{sz * num_wires, sz};

                 // return 2-D NumPy array
                 return py::array(py::buffer_info(
                     result.data(),  /* data as contiguous array  */
                     sz, /* size of one scalar        */
```

Reply:
good suggestion